DDCAD/include/DDCAD/Utilities.h

```diff
@@ -0,0 +1,76 @@
+//==========================================================================
+//  AIDA Detector description implementation
+//--------------------------------------------------------------------------
+// Copyright (C) Organisation europeenne pour la Recherche nucleaire (CERN)
+// All rights reserved.
+//
+// For the licensing terms see $DD4hepINSTALL/LICENSE.
+// For the list of contributors see $DD4hepINSTALL/doc/CREDITS.
+//
+// Author     : A. Sailer
+//
+//==========================================================================
+#ifndef DDCAD_UTILITIES_H
+#define DDCAD_UTILITIES_H
+
+#include <vector>
+
+#include <TGeoTessellated.h>
+#include <TGeoVector3.h>
+
+/// Namespace for the AIDA detector description toolkit
+namespace dd4hep {
+
+  /// Namespace for implementation details of the AIDA detector description toolkit
+  namespace cad  {
+
+    inline std::stringstream streamFacet(TGeoFacet const& facet,
+					 TGeoTessellated const& shape) {
+      using ::operator<<;
+      std::stringstream str;
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6,31,1)
+      str << "{";
+      for (int i = 0; i < facet.GetNvert(); ++i) {
+	str << shape.GetVertex(facet[i]);
+	if (i != facet.GetNvert() - 1)
+	  str << ", ";
+      }
+      str << "}";
+#else
+      str << facet;
+#endif
+      return str;
+    }
+
+    inline std::stringstream streamVertices(ROOT::Geom::Vertex_t const& v1,
+					    ROOT::Geom::Vertex_t const& v2,
+					    ROOT::Geom::Vertex_t const& v3) {
+      using ::operator<<;
+      std::stringstream str;
+      str << "{" << v1 << ", " << v2 <<  ", " << v3 << "}";
+      return str;
+    }
+
+    // Determine if the facet is degenerated by calculating its determinant
+    inline bool facetIsDegenerated(std::vector<ROOT::Geom::Vertex_t> const& vertices){
```

Comment:
@MarkusFrankATcernch 
Could you have a look at this implementation, please?

Reply:
Implementation look fine. I would perhaps make epsilon a bit bigger.
Typical values for such cutoffs in G4 are around 1e-10 to 1e-12.
Just in case....

Reply:
I took the 1e-20 from the previous implementation of ComputeNormal, though of course this is comparing different things to it.

https://github.com/root-project/root/blob/0bbefa75bb7c61991563089166864e6deb181412/geom/geom/src/TGeoTessellated.cxx#L78

Reply:
```
inline std::stringstream streamVertices(...) ....
```
Better would be:
```
inline std::ostream& streamVertices(std::ostream& os, ....)   {
  os << ......;
  return os;
}
```
This can then be fully integrated in the usage of std::cout etc.

Alternative: `toString` method:
```
inline std::string toString(const Object& o)   {
  use stringstresm like in your implementation
  return str.str();
}
```

Reply:
As this function is just used in one place at the moment I will leave it as is.